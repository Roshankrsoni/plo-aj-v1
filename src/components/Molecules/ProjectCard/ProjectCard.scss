.projectcard {
  border: 1px solid var(--grey5);
  padding: 1.5rem 1rem !important;
  border-radius: 0.25rem;
  width: 49%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  cursor: pointer;
}

.projectcard:hover {
  background-color: var(--grey6);
}

@media (max-width: 1200px) {
  .projectcard {
    width: 100%;
  }
  .projectcard:hover {
    background-color: transparent;
  }
}

.tags {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  grid-gap: 0.5rem;
}

.tag {
  font-size: 0.65rem !important;
  color: var(--grey6);
  font-weight: bold;
  background-color: var(--red);
  padding: 1px 0.5rem;
  border-radius: 0.25rem;
  width: 4rem;
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;
}

.description {
  padding-top: 1rem;
  max-height: 13rem;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 0.8125rem;
}

.projectlinks {
  display: flex;
}
.projectlinks .contactimg {
  width: 1.5rem;
}

.projectlinks a + a {
  margin-left: 1rem;
}

.projectcard .footer {
  display: flex;
  justify-content: space-between;
  padding-top: 1rem;
}

.projectstatus {
  display: flex;
  align-items: center;
}

.statuscircle {
  border-radius: 50%;
  height: 0.75rem;
  width: 0.75rem;
  margin-right: 0.5rem;
}
.projectstatus .label::first-letter {
  text-transform: uppercase;
}
.projectstatus.Progress .statuscircle {
  background-color: var(--orange);
}
.projectstatus.Complete .statuscircle {
  background-color: var(--green);
}
.projectstatus.Complete .label {
  color: var(--green);
}
.projectstatus.Progress .label {
  color: var(--orange);
}
