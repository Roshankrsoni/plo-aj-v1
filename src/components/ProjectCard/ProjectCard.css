.projectcard{
    border:1px solid #E0E0E0;
    padding:1.5rem 1rem !important;
    border-radius: 0.25rem;
    min-width: 20rem;
    width: 33.33%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    cursor: pointer;
}
.projectcard:hover{
   background-color: #f7f7f7;
}
.darkmode .projectcard{
    border:1px solid #4F4F4F;
    transition: border 0.5s ease-in-out;
}
.darkmode .projectcard:hover{
    background-color: #4F4F4F;
 }

.tags{
    display:flex;
   
    flex-wrap: wrap;
}

.tag{
    font-size: 0.65rem !important;
    color:#f2f2f2;
    font-weight: bold;
    background-color: #EB5757;
    padding:1px 0.5rem;
    border-radius: 20px;
    margin-right:0.5rem;
    width:4rem;
    text-align: center;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    cursor: pointer;
}
.tag + .tag:nth-child(4n + 2):not(:nth-child(2)),
.tag + .tag:nth-child(4n + 1):not(:nth-child(1)),
.tag + .tag:nth-child(4n){
    margin-top:0.5rem;
}
.showall .tag + .tag:nth-child(4n + 2):not(:nth-child(2)),
.showall .tag + .tag:nth-child(4n + 1):not(:nth-child(1)),
.showall .tag + .tag:nth-child(4n){
    margin-top:0;
}

.description{
    padding-top:1rem;
    max-height:13rem;
    overflow: hidden;
    text-overflow: ellipsis;
}

.projectlinks{
    display: flex;
}
.projectlinks .contactimg{
    width:1.5rem;
}

.projectlinks a + a{
   margin-left: 1rem;
}

.projectcard .footer{
display: flex;
justify-content: space-between;
padding-top:1rem;
}

.projectstatus{
display: flex;
align-items: center;
}

.statuscircle{
    border-radius: 50%;
    height:0.75rem;
    width:0.75rem;
    margin-right: 0.5rem;
}
.projectstatus .label::first-letter{
    text-transform: uppercase;
}
.projectstatus.Progress .statuscircle{
    background-color: #F2C94C;
}
.projectstatus.Complete .statuscircle{
    background-color: #27AE60;
}
.projectstatus.Complete .label{
    color: #27AE60;
}
.projectstatus.Progress .label{
    color: #F2C94C;
}