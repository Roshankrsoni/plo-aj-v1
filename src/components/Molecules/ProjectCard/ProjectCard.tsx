import React from "react";
import "./ProjectCard.scss";
import { ProjectDetails } from "../../../types/profileDataTypes";
import { FiExternalLink, FiGithub } from "react-icons/fi";

const ProjectCard = ({
  projectDetails,
}: {
  projectDetails: ProjectDetails;
}) => {
  return (
    <div className="projectcard">
      <div>
        <div className="h4 bold">{projectDetails.projectName}</div>
        <div className="tags">
          {projectDetails.tags?.map((ele, index) => {
            return (
              <div className="tag label" key={String(index)} title={ele}>
                {ele}
              </div>
            );
          })}
        </div>
        <div className="description">{projectDetails.description}</div>
      </div>
      <div className="footer">
        <div className="projectlinks">
          {projectDetails.github && (
            <a
              href={projectDetails.github}
              target="_blank"
              rel="noopener noreferrer"
            >
              <FiGithub className="contactimg" title="github link" />
            </a>
          )}
          {projectDetails.url && (
            <a
              href={projectDetails.url}
              target="_blank"
              rel="noopener noreferrer"
            >
              <FiExternalLink className="contactimg" title="application link" />
            </a>
          )}
        </div>
        <div className={`projectstatus ${projectDetails.status}`}>
          <div className="statuscircle" />
          <div className="label">{projectDetails.status}</div>
        </div>
      </div>
    </div>
  );
};

export default ProjectCard;
