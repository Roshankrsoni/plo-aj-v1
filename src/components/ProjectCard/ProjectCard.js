import React from 'react'
import './ProjectCard.css'
import github from '../../assets/github.svg';
import weblink from '../../assets/weblink.svg';

const ProjectCard = (props) => {
    return (
        <div className="projectcard">
            <div>
            <div className="header h4 bold">
                {props.data.projectName}
            </div>
            <div className="tags">
                {props.data.tags && props.data.tags.map((ele, index) => {
                    return (
                        <div className="tag label" key={index} title={ele}>
                            {ele}
                        </div>
                    )
                })}
            </div>
            <div className="description">
                {props.data.description}
            </div>
            </div>
            <div className="footer">
            <div className="projectlinks">
                { props.data.github && <a href={props.data.github}><img className="contactimg" title="view project on github" src={github} alt="github"/></a>}
                { props.data.url && <a href={props.data.url}><img className="contactimg" title="open app" src={weblink} alt="github"/></a> } 
            </div>
            <div className={`projectstatus ${props.data.status}`}>
                <div className="statuscircle"></div>
                <div className="label">{props.data.status}</div>
            </div>
            </div>
        </div>
    )
}

export default ProjectCard;