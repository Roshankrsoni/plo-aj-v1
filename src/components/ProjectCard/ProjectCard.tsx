import React from "react";
import "./ProjectCard.css";
import github from "../../assets/github.svg";
import weblink from "../../assets/weblink.svg";
import { ProjectDetails } from "../../Types/profileDataTypes";

const ProjectCard = ({
  projectDetails,
}: {
  projectDetails: ProjectDetails;
}) => {
  return (
    <div className="projectcard">
      <div>
        <div className="header h4 bold">{projectDetails.projectName}</div>
        <div className="tags">
          {projectDetails.tags &&
            projectDetails.tags.map((ele, index) => {
              return (
                <div className="tag label" key={index} title={ele}>
                  {ele}
                </div>
              );
            })}
        </div>
        <div className="description">{projectDetails.description}</div>
      </div>
      <div className="footer">
        <div className="projectlinks">
          {projectDetails.github && (
            <a
              href={projectDetails.github}
              target="_blank"
              rel="noopener noreferrer"
            >
              <img
                className="contactimg"
                title="view project on github"
                src={github}
                alt="github"
              />
            </a>
          )}
          {projectDetails.url && (
            <a
              href={projectDetails.url}
              target="_blank"
              rel="noopener noreferrer"
            >
              <img
                className="contactimg"
                title="open app"
                src={weblink}
                alt="github"
              />
            </a>
          )}
        </div>
        <div className={`projectstatus ${projectDetails.status}`}>
          <div className="statuscircle"></div>
          <div className="label">{projectDetails.status}</div>
        </div>
      </div>
    </div>
  );
};

export default ProjectCard;
