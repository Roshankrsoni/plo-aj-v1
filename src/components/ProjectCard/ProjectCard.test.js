import React from 'react';
import { shallow } from 'enzyme';
import ProjectCard from './ProjectCard';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<ProjectCard data={dummyData.projects[0]} />);
        expect(component).toMatchSnapshot();
    });
});