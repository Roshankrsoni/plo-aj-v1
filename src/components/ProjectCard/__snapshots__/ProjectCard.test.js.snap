// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="projectcard"
>
  <div>
    <div
      className="header h4 bold"
    >
      Guest Tracker
    </div>
    <div
      className="tags"
    >
      <div
        className="tag label"
        key="0"
        title="React Js"
      >
        React Js
      </div>
      <div
        className="tag label"
        key="1"
        title="Firebase"
      >
        Firebase
      </div>
    </div>
    <div
      className="description"
    >
      A React App to manage guests or organize an event. The idea behind creation of this app was to manage the guests or events in this covid pandemic situation because of the limition on the number of guests allowed by the government authorities.
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="projectlinks"
    >
      <a
        href="https://github.com/Anoopjadhav/guest-tracker"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="github"
          className="contactimg"
          src="github.svg"
          title="view project on github"
        />
      </a>
      <a
        href="https://guesttracker.netlify.app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="github"
          className="contactimg"
          src="weblink.svg"
          title="open app"
        />
      </a>
    </div>
    <div
      className="projectstatus Complete"
    >
      <div
        className="statuscircle"
      />
      <div
        className="label"
      >
        Complete
      </div>
    </div>
  </div>
</div>
`;
